### PR TITLE
CI: add Python 3.11 to the test matrix

### LINKED ISSUE #2592: ENH: Support Python 3.11
Python 3.11 is expected to be released as stable around October 24th, with many new features including:

> - [PEP 657](https://www.python.org/dev/peps/pep-0657/) -- Include Fine-Grained Error Locations in Tracebacks
> - [PEP 654](https://www.python.org/dev/peps/pep-0654/) -- Exception Groups and except*
> - [PEP 673](https://www.python.org/dev/peps/pep-0673/) -- Self Type
> - [PEP 646](https://www.python.org/dev/peps/pep-0646/)-- Variadic Generics
> - [PEP 680](https://www.python.org/dev/peps/pep-0680/)-- tomllib: Support for Parsing TOML in the Standard Library
> - [PEP 675](https://www.python.org/dev/peps/pep-0675/)-- Arbitrary Literal String Type
> - [PEP 655](https://www.python.org/dev/peps/pep-0655/)-- Marking individual TypedDict items as required or potentially-missing
> - [bpo-46752](https://bugs.python.org/issue46752)-- Introduce task groups to asyncio
> - The [Faster Cpython Project](https://github.com/faster-cpython) is already yielding some exciting results: this version of CPython 3.11 is ~ 19% faster on the geometric mean of the [PyPerformance benchmarks](https://speed.python.org/), compared to 3.10.0.

The state of the Python 3.11 ecosystem is already great:

- ✅ The regular scientific computing stack is in excellent shape. NumPy and Pandas support Python 3.11 and have wheels available, as do SciPy, Matplotlib and Seaborn. Tools like, Black and pre-commit and pytest also support Python 3.11.
- The geospatial stack is build on top of those, and now also gaining Python 3.11 support very quickly
  - 🚀 fiona the [CI passes](https://github.com/Toblerity/Fiona/pull/1147) for 3.11, wheel building is being worked on. See tracking issue: https://github.com/Toblerity/Fiona/issues/1087
  - ✅ shapely just released version [1.8.5](https://github.com/shapely/shapely/releases/tag/1.8.5), with 3.11 wheels on PyPI!
  - ✅ pyproj supports Python 3.11 and has wheels on PyPI
  - ✅ SQLAlchemy idem dito
  - ✅ rtree just released their [1.0.1](https://github.com/Toblerity/rtree/releases/tag/1.0.1) release which include Python 3.11 support and wheels!
  - ✅ geopy has a version-independent Python wheel available

GeoPandas is the glue that connects the whole geospatial Python stack. I think it would very well be possible to have full Python 3.11 support in GeoPandas before the official release! (currently planned for October 24th). Would that be a target to aim for?

### PR DESCRIPTION
**Pull request by martinfleis**
With pyogrio being rebuilt for 3.11 (~Fiona not yet~), we should have all we need to test geopandas with Python 3.11.

~This env also excludes pygeos and uses shapely 2.~ (does not resolve)

Closes #2592

### CODE REVIEW

File: ci/envs/311-latest-conda-forge.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - python=3.11
+  # required
+  - pandas
+  - shapely
+  - fiona
+  - tiledb=2.11.3
```
Comment (jorisvandenbossche) [inline ci/envs/311-latest-conda-forge.yaml:10]: I suppose this isn't actually needed anymore

Comment (martinfleis) [reply ci/envs/311-latest-conda-forge.yaml]: Yeah, doesn't seem so. Removed from all envs.

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Comment by EwoutH** [comment]
Awesome!

**Pull request merged**